### PR TITLE
Fixed bugs

### PR DESCRIPTION
**Pull request by ashwini-ad**

### CODE REVIEW

File: python/awaazde/base.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def list(self, **kwargs):
         """
         data = {}
         if kwargs:
-            data = {'params': kwargs}
+            data = {'params': kwargs.pop('params')}
```
Comment (kapilrupavatiya) [inline python/awaazde/base.py:49]: @ashwini-ad Can't we directly pass params in the kwargs, I didn't see any need to pass it as a params key in kwargs argument.

Also can you please test what happens if we pass an empty dictionary in params of her request, If that works then you can also remove this if condition?

**Review by kapilrupavatiya — CHANGES_REQUESTED** [review]

File: python/awaazde/apis.py [hunk 2]
```diff
@@ -82,10 +82,12 @@ def upload(self, file_path):
 
         return self._client.post(upload_url, **data)
 
-    def create_bulk(self, data):
+    def create_bulk(self, data,**kwargs):
         """
         This will create new object
         """
         bulk_url = self.url + "create_bulk/"
+        transform_using_template = kwargs.pop('transform_using_template')
```
[the remark this answers is not in the document]
Comment (kapilrupavatiya) [reply python/awaazde/apis.py:92]: @ashwini-ad Based on this implementation, We are passing transform_using_template every time in request and the value of this will change based on passed data in kwargs. So I feel it's not a proper use of kwargs. Ideally, after the data parameter, You can pass another parameter called "transform_using_template" and specify the default value of False. So that it will be an explicit parameter of this method and everyone can see the same.

File: python/awaazde/base.py [hunk 3]
```diff
@@ -131,14 +127,15 @@ def create_bulk_in_chunks(self, data, **kwargs):
...
-            response += self.create_bulk(data_chunk)
+            response += self.create_bulk(data_chunk, **kwargs)
         return response
 
     def list_depaginated(self, params=None):
         """
             Gets all messages from awaazde API based on the filters passed
         """
         data = []
+        print(params)
```
Comment (kapilrupavatiya) [inline python/awaazde/base.py:138]: @ashwini-ad please remove this print statement.

File: python/list_and_match_messages.py [hunk 4]
```diff
@@ -33,22 +40,29 @@ def parse_arguments():
     """
         Step 2 : Send an api request to get list of messages based on filters
     """
-    params = {'page': APIConstants.DEFAULT_BULK_CREATE_LIMIT,
-              'fields': (CommonConstants.PHONE_NUMBER_FIELD, CommonConstants.ID_FIELD, CommonConstants.SEND_ON_FIELD),
-              "filters": dict(list(map(str.strip, s.split('='))) for s in args.params)}
+    params = {'fields': ','.join(
+        [CommonConstants.PHONE_NUMBER_FIELD, CommonConstants.ID_FIELD, CommonConstants.SEND_ON_FIELD]),
+    }
+    filters = dict(list(map(str.strip, s.split('='))) for s in args.params)
```
Comment (kapilrupavatiya) [inline python/list_and_match_messages.py:46]: @ashwini-ad If we are splitting params with "=" sign then it will work when all params will have "=" sign only.

Based on your description of params argument, You maintain that >= and <= will also allow, So in >= and <= cases how this split will work. I feel it will generate the wrong filters directory.

Comment (ashwini-ad) [reply python/list_and_match_messages.py:46]: We would not use signs here for greater than, but use "__gt" and "__lt" like we do in django queries

File: python/awaazde/apis.py [hunk 5]
```diff
@@ -82,10 +82,13 @@ def upload(self, file_path):
...
-    def create_bulk(self, data):
+    def create_bulk(self, data, transform_using_template=False, **kwargs):
         """
         This will create new object
         """
         bulk_url = self.url + "create_bulk/"
-        data = {'json': data}
-        return self._client.post(bulk_url, **self._append_headers(data))
\ No newline at end of file
+        # We need to pop send "transform_using_template" as a separate parameter other than "data"
```
Comment (kapilrupavatiya) [inline python/awaazde/apis.py:90]: @ashwini-ad please remove these comments and please add a method parameter explainatation.

File: python/awaazde/base.py [hunk 6]
```diff
@@ -116,9 +114,7 @@ def _append_headers(self, data, append_content_type=True):
         data['headers'] = headers
         return data
 
-
-
-    def create_bulk_in_chunks(self, data, **kwargs):
+    def create_bulk_in_chunks(self, data, transform_using_template=False, **kwargs):
         """
         Create objects in chunks based on limit if present, takes DEFAULT_BULK_CREATE_LIMIT as default.
         :param Data: Data to create. eg: if messages: [{phone_number:8929292929,send_on:"",tag1:"tag_number1",templatelanguage:23,language:"hi"}]
```
Comment (kapilrupavatiya) [inline python/awaazde/base.py:120]: @ashwini-ad please change this parameter description and make it consistent with current method parameter.

File: python/list_and_match_messages.py (hunk 4 above)
Comment (kapilrupavatiya) [reply python/list_and_match_messages.py:46]: @ashwini-ad in that case please change the example you are providing the same in above of the method. Make it consistance.
